### ISSUE TITLE
Error getting zone records

### ISSUE THREAD

**Issue by johanjeppsson** [caddy-dns/loopia#1]
I am very happy that someone had already done some work to get the loopia DNS working with caddy. :smile:

I try to get it up and running but run into some errors. I'm not to familiar with Caddy, so I'm not sure if I'm misconfiguring something or if there is some issue with the module.

My configuration looks like this

```
*.<mydomain>.se {
	@sub host sub.<mydomain>.se
	handle @sub {
		reverse_proxy localhost:8888
	}

	handle {
    	    # Set this path to your site's directory.
    	    root * /usr/share/caddy
    	    # Enable the static file server.
    	    file_server
	}
    tls {
        issuer acme {
            propagation_timeout "15m0s"
            dns loopia {
                    username "<api-user>@loopiaapi"
                    password "<api-user-password>"
            }
        }
    }
}
```
I get the following error in the log

```
{"level":"error","ts":1646728143.6791425,"logger":"tls.obtain","msg":"will retry","error":"[*.<mydomain>.se] Obtain: [*.<mydomain>.se] solving challenges: presenting for challenge: adding temporary record for zone <mydomain>.se.: unexpected error getting zone records: Fault(623): Calling parameters do not match signature (order=https://acme-staging-v02.api.letsencrypt.org/acme/order/...) (ca=https://acme-staging-v02.api.letsencrypt.org/directory)","attempt":4,"retrying_in":300,"elapsed":305.132993075,"max_duration":2592000}
```

The API-user has the permission listed in the readme, and it looks like it's being called, but I'm not sure how to debug it further. Any ideas on what I can try?

**Comment by kmpm** [comment]
I have a sneeky suspicion that the wildcard DNS you are asking for doesn't work.
Could you please try something like "host.mydomain.se" instead of "*.mydomain.se"

**Comment by johanjeppsson** [comment]
Without the wildcard it works, but that works without the loopia-dns module as well. My goal, which might be unnecessarily complicated, was to get a wildcard certificate from letsencrypt instead of a separate certificate for each subdomain. From what I've read in the caddy documentation, this requires a [DNS challenge](https://caddyserver.com/docs/automatic-https#wildcard-certificates), which requires a DNS provider.
But maybe I should just keep things simple and use separate certificates for each subdomain instead. I should be fine with the limits imposed by letsencrypt for now.

**Comment by johanjeppsson** [comment]
It looks like the loopia-API has changed recently. According to [this blogpost](https://techblog.se/2020/01/19/automating-certbot-dns-validation-with-loopiadns-lets-encrypt/):

>I den nya versionen så kan inte kunden skicka med ett kundnummer som en parameter, vilket gör att de som skickar med detta, eller skickar med en tom sträng får ett felmeddelande (oftast error 623).
Undantaget här är återförsäljare som fortfarande behöver skicka med kundnummer när de gör anrop mot sina kunders konton.

This seems to match the error code that I was getting. I guess that if this is the culprit, it's libdns/loopia that needs to be updated?

**Comment by kmpm** [comment]
Thanks for testing the explicit subdomain.
I will look into it this weekend.

**Comment by kmpm** [comment]
Fixed in 1e79ab75e5ea3b330b5c0074c03548e4c5b0415f  tagged as v0.0.2
Use with `xcaddy build --with github.com/caddy-dns/loopia@v0.0.2`.

And I do have successfully used it with wildcard domains as in your example.

**Comment by johanjeppsson** [comment]
Works for me as well! Thank you! :)
